src/primitives.rs

```diff
@@ -333,6 +340,168 @@ where
     }
 }
 
+impl<T> Versionize for VecDeque<T>
```

Comment:
This is almost like implementation for Vec<T>. Can we do this with a macro to avoid duplicating code ?